optaplanner-core/src/main/java/org/optaplanner/core/config/heuristic/selector/entity/pillar/SubPillarConfigPolicy.java

```diff
@@ -0,0 +1,106 @@
+        this.minimumSubPillarSize = minimumSubPillarSize;
+        this.maximumSubPillarSize = maximumSubPillarSize;
+        validateSizes();
+        if (entityComparator == null) {
+            throw new IllegalStateException("entityComparator must not be null.");
+        }
+        this.entityComparator = entityComparator;
+    }
+
+    private SubPillarConfigPolicy() {
+        this.subPillarEnabled = false;
+        this.minimumSubPillarSize = -1;
+        this.maximumSubPillarSize = -1;
+        this.entityComparator = null;
+    }
+
+    public static SubPillarConfigPolicy withoutSubpillars() {
+        return new SubPillarConfigPolicy();
+    }
+
+    public static SubPillarConfigPolicy withSubpillars(int minSize, int maxSize) {
+        return new SubPillarConfigPolicy(minSize, maxSize);
+    }
+
+    public static SubPillarConfigPolicy withSubpillarsUnlimited() {
+        return withSubpillars(1, Integer.MAX_VALUE);
+    }
+
+    public static SubPillarConfigPolicy sequential(int minSize, int maxSize, Comparator<?> entityComparator) {
+        return new SubPillarConfigPolicy(minSize, maxSize, entityComparator);
+    }
+
+    public static SubPillarConfigPolicy sequentialUnlimited(Comparator<?> entityComparator) {
+        return sequential(1, Integer.MAX_VALUE, entityComparator);
+    }
+
+    private void validateSizes() {
+        if (minimumSubPillarSize < 1) {
+            throw new IllegalStateException("The selector (" + this
+                    + ")'s minimumPillarSize (" + minimumSubPillarSize
+                    + ") must be at least 1.");
+        }
+        if (minimumSubPillarSize > maximumSubPillarSize) {
+            throw new IllegalStateException("The minimumPillarSize (" + minimumSubPillarSize
+                    + ") must be at least maximumSubChainSize (" + maximumSubPillarSize + ").");
+        }
+    }
+
+    public boolean isSubPillarEnabled() {
+        return subPillarEnabled;
+    }
+
+    /**
+     * @return Less than 1 when {@link #isSubPillarEnabled()} false.
+     */
+    public int getMinimumSubPillarSize() {
+        return minimumSubPillarSize;
+    }
+
+    /**
+     * @return Less than 1 when {@link #isSubPillarEnabled()} false.
+     */
+    public int getMaximumSubPillarSize() {
+        return maximumSubPillarSize;
+    }
+
+    /**
+     * @return Only present if the subpillars are to be treated as sequential. Never present if
+     * {@link #isSubPillarEnabled()} is false.
+     */
+    public Optional<Comparator<?>> getEntityComparator() {
+        return Optional.ofNullable(entityComparator);
+    }
+
+    @Override
+    public String toString() {
+        return getClass().getSimpleName() + "{" +
+                "entityComparator=" + entityComparator +
+                ", maximumSubPillarSize=" + maximumSubPillarSize +
+                ", minimumSubPillarSize=" + minimumSubPillarSize +
+                ", subPillarEnabled=" + subPillarEnabled +
+                '}';
```

Comment:
optaplanner-core doesn't use verbose toStrings.
Our philophy (which works really well in practice with log messages) is that toStrings() should not be used to debug - the IntelliJ/VsCode debugger has plenty of ways to look at the fields of a instance.
Instead, they must give an indication of what it is, in essence. Less is more. See toStrings() on other classes (ChangeMoveSelector, ChangeMove, FilterConstraintStream, ...)

PS: why the + '}' (single quotes)?

Reply:
OK. In this case, I'll just remove the `toString()`.
The single quotes are auto-generated by IntelliJ. I assume it's because "}" is a single character and, as such, using a `String` for it is wasteful.